networking/p2p/src/main/java/tech/pegasys/artemis/networking/p2p/network/P2PNetwork.java

```diff
@@ -88,6 +89,15 @@
    */
   Optional<String> getEnr();
 
+  /**
+   * Updates the value of or creates a field for the Ethereum Node Record (ENR) for the local node,
+   * if one exists.
+   *
+   * @param fieldName
+   * @param value
+   */
+  void updateCustomENRField(String fieldName, Bytes value);
```

Comment:
(nit) Maybe just:
```suggestion
  void setENRField(String fieldName, Bytes value);
```

Reply:
updateCustomENRField is not the best, but there can be multiple different custom fields, so set wouldn't be too correct either. I'm undecided.